### PR TITLE
修复编译过程中的 DFS_V2报错、FAL警告

### PR DESCRIPTION
**Pull request by yuqingli05**
## 拉取/合并请求描述：(PR description)

[
并确认并列出已经在什么情况或板卡上进行了测试。
And confirm in which case or board has been tested. -->
自己板子

#### 为什么提交这份PR (why to submit this PR)
当定义了RT_USING_DFS_V2但未定义RT_RT_USING_DFS_DEVFS时，编译报错
当定义了RT_USING_DFS_V2但未定义RT-USING_SMART时，会发生错误
fal_rtt.c 缺少头，提醒函数未定义

#### 你的解决方案是什么 (what is your solution)
修改源码

#### 请提供验证的bsp和config (provide the config and bsp) 
 [ 
[config.txt](https://github.com/user-attachments/files/18203351/config.txt)
]

- BSP: bsp\stm32\stm32f103-atk-nano

- action:

]

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [ ] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [ ] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [ ] 代码是高质量的 Code in this PR is of high quality
- [ ] 已经使用[formatting](https://github.com/mysterywolf/formatting) 等源码格式化工具确保格式符合[RT-Thread代码规范](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_cn.md) This PR complies with [RT-Thread code specification](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_en.md)
- [ ] 如果是新增bsp, 已经添加ci检查到[.github/workflows/bsp_buildings.yml](workflows/bsp_buildings.yml)  详细请参考链接[BSP自查](https://www.rt-thread.org/document/site/#/rt-thread-version/rt-thread-standard/development-guide/bsp-selfcheck/bsp_selfcheck)

### CODE REVIEW

File: components/drivers/core/device.c [hunk 1]
```diff
@@ -31,9 +31,9 @@
 #include <rtdevice.h> /* for wqueue_init */
 #endif /* RT_USING_POSIX_DEVIO */
 
-#ifdef RT_USING_DFS_V2
+#if((defined RT_USING_DFS_V2) && (defined RT_USING_DFS_DEVFS))
```
Comment (mysterywolf) [inline components/drivers/core/device.c:34]:
```suggestion
#if defined (RT_USING_DFS_V2) && defined (RT_USING_DFS_DEVFS)
```

File: components/drivers/core/device.c [hunk 2]
```diff
@@ -84,7 +84,7 @@ rt_err_t rt_device_register(rt_device_t dev,
     rt_wqueue_init(&(dev->wait_queue));
 #endif /* RT_USING_POSIX_DEVIO */
 
-#ifdef RT_USING_DFS_V2
+#if((defined RT_USING_DFS_V2) && (defined RT_USING_DFS_DEVFS))
```
Comment (mysterywolf) [inline components/drivers/core/device.c:87]:
```suggestion
#if defined (RT_USING_DFS_V2) && defined (RT_USING_DFS_DEVFS)
```

**Review by mysterywolf — CHANGES_REQUESTED** [review]

**Comment by yuqingli05** [comment]
@myshowtogo 不好意思，[42e7546](https://github.com/RT-Thread/rt-thread/pull/9806/commits/42e7546ebfd97777a95fca457db3d8e82000073e) 和 [9d28f83](https://github.com/RT-Thread/rt-thread/pull/9806/commits/9d28f831609092ad4f267da2cc5bc16e23b3f69f) 不属于PR。这是我错误的提交到这个分支导致的。可以忽略，后面我会单独提交PR来套路这个问题。

**Comment by BernardXiong** [comment]
请都独立来提交PR，并且提交的commit请提供更详细的信息，谢谢

**Comment by yuqingli05** [comment]
从新提交 https://github.com/RT-Thread/rt-thread/pull/9826。关闭

**Pull request closed without merging**